ProjectName/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  NetworkAPIProtocol.swift
+//
+
+import RxSwift
+import Alamofire
+import RxAlamofire
+
+protocol NetworkAPIProtocol {
+
+    func performRequest<T: Decodable>(_ configuration: RequestConfiguration, for type: T.Type) -> Single<T>
+}
+
+extension NetworkAPIProtocol {
+
+    func request<T: Decodable>(
+        session: Session,
+        configuration: RequestConfiguration,
+        decoder: JSONDecoder
+    ) -> Single<T> {
+        return session.rx.request(
+            configuration.method,
+            configuration.url,
+            parameters: configuration.parameters,
+            encoding: configuration.encoding,
+            headers: configuration.headers,
+            interceptor: configuration.interceptor
+        )
+        .responseData()
+        .observe(on: ConcurrentDispatchQueueScheduler(qos: .background))
+        .flatMap { _, data -> Observable<T> in
+            Observable.create { observer in
+                do {
+                    let decodable = try decoder.decode(T.self, from: data)
+                    observer.on(.next(decodable))
+                } catch {
+                    observer.on(.error(error))
+                }
+                observer.on(.completed)
+                return Disposables.create()
+            }
+        }
+        .observe(on: MainScheduler.instance)
```

Comment:
Agree with @phongvhd93 and @minhnimble, I think we can remove `observe` operators here, and the developer can control more by bringing the observe part into the usecase
```suggestion
        .flatMap { _, data -> Observable<T> in
            Observable.create { observer in
                do {
                    let decodable = try decoder.decode(T.self, from: data)
                    observer.on(.next(decodable))
                } catch {
                    observer.on(.error(error))
                }
                observer.on(.completed)
                return Disposables.create()
            }
        }
```